File: UA_SD-master/Practices/practica 1/src/cliente.py
import socket
import sys

def Main():
    
    if len(sys.argv) < 3:
        print("Missing parameters.")
        exit()

    host = sys.argv[1]
    port = int(sys.argv[2])
    
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.connect((host, port))

    while True:

        command = input('Insert command: ')

        if command == "exit":
            s.send(command.encode('utf-8'))
            response = s.recv(1024).decode('utf-8')

            if "EXIT" == response:
                break

        s.send(command.encode('utf-8'))
        response = s.recv(1024).decode('utf-8')
        print(response)

    s.close()


Main()

File: UA_SO-master/Practica2/cliente.c
// Elvi Mihai Sabau Sabau
#include <stdlib.h>
#include <unistd.h>
#include <arpa/inet.h>
#include <sys/types.h>
#include <netinet/in.h>
#include <stdio.h>
#include <sys/socket.h>

void main(int argc, char const *argv[])
{
  // Declaramos las variables, los descriptores, el buffer...
  struct sockaddr_in servidor;
  int fd;
  char buffer[2048] = {0};

  const char *comando = argv[2];

  // Creamos el socket.
  fd = socket(AF_INET, SOCK_STREAM, 0);
  if (fd == -1)
  {
    perror("Error creando el socket");
    exit(1);
  }

  // Especificamos los parametros del socket.
  servidor.sin_family = AF_INET;
  servidor.sin_port = htons(9999);
  servidor.sin_addr.s_addr = inet_addr(argv[1]);

  // Abrimos una nueva conxión.
  if (connect(fd, (struct sockaddr *)&servidor, sizeof(servidor)) == -1)
  {
    perror("Error realizando la conexión.");
    exit(1);
  }

  // Enviamos los comandos, y recibimos los datos.
  if (send(fd, comando, sizeof(comando), 0) == -1)
  {
    perror("Error enviando datos.");
    exit(1);
  }

  if (recv(fd, buffer, sizeof(buffer), 0) == -1)
  {
    perror("Error recibiendo datos.");
    exit(1);
  }

  printf("----------SALIDA DEL COMANDO %s ----------\n%s", comando, buffer);

  exit(0);
}
